packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -687,9 +747,18 @@ class TransformLayer extends OffsetLayer {
   ///
   /// The [transform] property must be non-null before the compositing phase of
   /// the pipeline.
-  Matrix4 transform;
+  Matrix4 get transform => _transform;
+  Matrix4 _transform;
+  set transform(Matrix4 value) {
+    if (value == _transform)
+      return;
+    _transform = value;
+    _invertedTransform ??= new Matrix4.zero();
+    _transform?.copyInto(_invertedTransform)?.invert();
```

Comment:
shouldn't this just set _invertedTransform to null? It seems weird that we aggressively precompute the second time, but not the first time, that this is set.